extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/GraphiteEventConverter.java

```diff
@@ -0,0 +1,241 @@
+    this.whiteListDimsMapper = graphiteConfig.getWhiteListDimsMapper() == null ? DEFAULT_DIM_WHITE_LIST_MAPPER : graphiteConfig.getWhiteListDimsMapper();
+    this.ignoreServiceName = graphiteConfig.getIgnoreServiceName();
+    this.ignoreHostname = graphiteConfig.getIgnoreHostname();
+    this.namespacePrefix = graphiteConfig.getNamespacePrefix();
+  }
+
+  public GraphiteEventConverter(
+      @NotNull String namespacePrefix,
+      @NotNull boolean ignoreHostname,
+      @NotNull boolean ignoreServiceName,
+      SortedMap<String, Set<String>> whiteListDimsMapper
+  )
+  {
+    this.whiteListDimsMapper = whiteListDimsMapper == null ? DEFAULT_DIM_WHITE_LIST_MAPPER : whiteListDimsMapper;
+    this.ignoreServiceName = ignoreServiceName;
+    this.ignoreHostname = ignoreHostname;
+    this.namespacePrefix = namespacePrefix;
+  }
+
+  public boolean isIgnoreHostname()
+  {
+    return ignoreHostname;
+  }
+
+  public boolean isIgnoreServiceName()
+  {
+    return ignoreServiceName;
+  }
+
+  public String getNamespacePrefix()
+  {
+    return namespacePrefix;
+  }
+
+
+  /*
+  * The event is considered in white list if and only if it has a prefix key for dims mapping and a valid metric type
+  * */
+  private boolean isInWhiteList(ServiceMetricEvent event)
+  {
+    return getPrefixKey(event.getMetric(), whiteListDimsMapper) != null;
+  }
+
+
+  /**
+   * @param key       the metric name to lookup
+   * @param whiteList
+   *
+   * @return <tt>null</tt> if the key does not match with any of the prefixes keys in @code metricsWhiteList,
+   * or the prefix in @code whiteListDimsMapper
+   */
+  private String getPrefixKey(String key, SortedMap<String, ?> whiteList)
+  {
+    String prefixKey = null;
+    if (whiteList.containsKey(key)) {
+      return key;
+    }
+    SortedMap<String, ?> headMap = whiteList.headMap(key);
+    if (!headMap.isEmpty() && key.startsWith(headMap.lastKey())) {
+      prefixKey = headMap.lastKey();
+    }
+    return prefixKey;
+  }
+
+  /**
+   * Returns a {@link List} of the white-listed dimension's values to send.
+   * The list is order is the same as the order of dimensions {@code whiteListDimsMapper}
+   *
+   * @param event the event for which will filter dimensions
+   *
+   * @return {@link List}  of the filtered dimension values to send or <tt>null<tt/> if the event is not in the white list
+   */
+  private List<String> getOrderedDimValues(ServiceMetricEvent event)
+  {
+    String prefixKey = getPrefixKey(event.getMetric(), whiteListDimsMapper);
+    if (prefixKey == null) {
+      return null;
+    }
+    ImmutableList.Builder<String> outputList = new ImmutableList.Builder();
+    Set<String> dimensions = whiteListDimsMapper.get(prefixKey);
+    if (dimensions == null) {
+      return Collections.emptyList();
+    }
+    for (String dimKey : dimensions) {
+      String dimValue = (String) event.getUserDims().get(dimKey);
+      if (dimValue != null) {
+        outputList.add(dimValue);
```

Comment:
Well first i thought of a selective inclusive to know exactly what we are emitting, let say tomorrow we get a new combination of metrics/dims, you will be getting events that you might don't like.
So As a solution to that we can either add a parameter to turn filtering off or read such map as input.
I am not very familiar with graphite, so please feel free to let me know if this doesn't make sense at all